app/models/current_user.rb

```diff
@@ -0,0 +1,37 @@
+# get the current User object from the Rails request object
+# Intended to be used as CurrentUser.for(request)
+class CurrentUser
+  attr_reader :request
+
+  def initialize(request)
+    @request = request
+  end
+
+  def self.for(request)
+    new(request).user_object
+  end
+
+  def user_object
+    @user_object ||= begin
+      if user_id.present?
+        webauth_user
+      else
+        anonymous_user
+      end
+    end
+  end
+
+  private
+
+  def webauth_user
+    User.find_or_create_by(webauth: user_id)
+  end
+
+  def anonymous_user
+    User.new(ip_address: request.remote_ip)
```

Comment:
Do we currently have any use for this up_address passing for this anonymous user object?